pkg/querier/tenantfederation/merge_queryable.go

```diff
@@ -144,23 +147,70 @@ type stringSliceFunc func(storage.Querier) ([]string, storage.Warnings, error)
 // the output of the stringSliceFunc to be sorted, as results of LabelValues
 // are not sorted.
 //
-// TODO: Consider running stringSliceFunc calls concurrently
+
+type stringSliceFuncResult struct {
+	Result   []string
+	Warnings storage.Warnings
+}
+
```

Comment:
**suggestion:** Consider moving this struct definition above the preceding comment since the comment itself pertains to the `mergeDistinctStringSlice` function.

It would also be nice to update the comment above to note this function runs each querier in parallel.

Reply:
Good point